pkg/controller/machine/machine_controller.go

```diff
@@ -221,17 +236,77 @@ func (r *ReconcileMachine) deleteNode(ctx context.Context, cluster *clusterv1.Cl
 	// Otherwise, proceed to get the remote cluster client and get the Node.
 	remoteClient, err := remote.NewClusterClient(r.Client, cluster)
 	if err != nil {
-		klog.Errorf("Error creating a remote client for cluster %q while deleting Machine %q, won't retry: %v",
+		klog.Errorf("error creating a remote client for cluster %q while deleting Machine %q, won't retry: %v",
 			cluster.Name, name, err)
 		return nil
 	}
 
 	corev1Remote, err := remoteClient.CoreV1()
 	if err != nil {
-		klog.Errorf("Error creating a remote client for cluster %q while deleting Machine %q, won't retry: %v",
+		klog.Errorf("error creating a remote client for cluster %q while deleting Machine %q, won't retry: %v",
 			cluster.Name, name, err)
 		return nil
 	}
 
 	return corev1Remote.Nodes().Delete(name, &metav1.DeleteOptions{})
 }
+
+func (r *ReconcileMachine) drainNode(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	var kubeClient kubernetes.Interface
```

Comment:
it'd be nice to keep the remote/local cluster client logic in its own func and just pass the client here

Reply:
Yes, I think once we've settled on a particular workflow there is room to consolidate some client building.  You'll probably need the same remote client for drain and delete, so we can combine later.